tests/Schema/TableTest.php

```diff
@@ -1755,4 +1759,121 @@ public function testInvalidPrimaryKeyIndex(): void
+
+        self::assertCount(3, $foreignKeys);
+        self::assertForeignKeyMatches($foreignKeys[0], ['t_id'], ['id'], 't1');
+        self::assertForeignKeyMatches($foreignKeys[1], ['t_id'], ['id'], 't2');
+        self::assertForeignKeyMatches($foreignKeys[2], ['t_id'], ['second_id'], 't2');
+    }
+
+    public function testAutoGeneratedForeignKeyNamesForConflictingReferencesUsingConstructor(): void
+    {
+        $table = Table::editor()
+            ->setUnquotedName('t')
+            ->setColumns(
+                Column::editor()
+                    ->setUnquotedName('id')
+                    ->setTypeName(Types::INTEGER)
+                    ->create(),
+                Column::editor()
+                    ->setUnquotedName('t_id')
+                    ->setTypeName(Types::INTEGER)
+                    ->create(),
+            )
+            ->setForeignKeyConstraints(
+                ForeignKeyConstraint::editor()
+                    ->setUnquotedReferencingColumnNames('t_id')
+                    ->setUnquotedReferencedTableName('t1')
+                    ->setUnquotedReferencedColumnNames('id')
+                    ->create(),
+                ForeignKeyConstraint::editor()
+                    ->setUnquotedReferencingColumnNames('t_id')
+                    ->setUnquotedReferencedTableName('t2')
+                    ->setUnquotedReferencedColumnNames('id')
+                    ->create(),
+                ForeignKeyConstraint::editor()
+                    ->setUnquotedReferencingColumnNames('t_id')
+                    ->setUnquotedReferencedTableName('t2')
+                    ->setUnquotedReferencedColumnNames('second_id')
+                    ->create(),
+                ForeignKeyConstraint::editor()
+                    ->setUnquotedReferencingColumnNames('t_id')
+                    ->setUnquotedReferencedTableName('t2')
+                    ->setUnquotedReferencedColumnNames('id')
+                    ->create(),
+                ForeignKeyConstraint::editor()
+                    ->setUnquotedReferencingColumnNames('t_id')
+                    ->setUnquotedReferencedTableName('t1')
+                    ->setUnquotedReferencedColumnNames('id')
+                    ->create(),
+            )
+            ->create();
+
+        /** @var list<ForeignKeyConstraint> $foreignKeys */
+        $foreignKeys = array_values($table->getForeignKeys());
+
+        self::assertCount(3, $foreignKeys);
+        self::assertForeignKeyMatches($foreignKeys[0], ['t_id'], ['id'], 't1');
+        self::assertForeignKeyMatches($foreignKeys[1], ['t_id'], ['id'], 't2');
+        self::assertForeignKeyMatches($foreignKeys[2], ['t_id'], ['second_id'], 't2');
+    }
+
+    /**
+     * Asserts that a single foreign key matches the expected column names and referenced table.
+     *
+     * @param string[] $expectedReferencingColumns
+     * @param string[] $expectedReferencedColumns
+     */
+    private static function assertForeignKeyMatches(
```

Comment:
I would try to avoid this code. This is unnecessary boilerplate. If we shift gears and only introduce the deprecation, we will only need to test if the deprecation is triggered w/o actually testing the final state of the object (this is already done in other existing tests).